core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -35,6 +37,8 @@
  * An abstraction over files and classpath resources aimed at encapsulating all the complexity of generating
  * a path that the Docker daemon is about to create a volume mount for.
  */
+@EqualsAndHashCode
```

Comment:
there are a few lazy fields, IMO only `path` should be included

Reply:
maybe `path` and `forcedFileMode`? BTW `resourcePath` is not used as a cache, so it's recalculated every time and could be a local variable.

Reply:
re `resourcePath` - yeah, that's a leftover

re `forcedFileMode` - but then a Map will have two items for the same path but different file modes.

Reply:
How's that?

Reply:
if you include `forcedFileMode` into equals/hashCode then it means that it will be a differentiator for the object.

Reply:
Oh, I meant: What do you think of my recent push ;-)

Reply:
ooops :D Ok, let's continue the discussion of `forceFileMode` in https://github.com/testcontainers/testcontainers-java/pull/2144#discussion_r353196717